js/ui/gantt/ui.gantt.js

```diff
@@ -551,6 +562,9 @@ class Gantt extends Widget {
             case "resourceAssignments":
                 this._refreshDataSource("resourceAssignments");
                 break;
+            case "treeListColumns":
+                this._setTreeListOption("columns", args.value);
```

Comment:
Do not use args.value here because user can change inner property. For example after calling gantt.option("treeListColumns[0].visible", false) args,name will be "treeListColumns, args,fullName will be "treeListColumns[0].visible", args.value will be false and this handler will be wrong.